apps/andi/lib/andi_web/live/ingestion_live_view/metadata/select_dataset_modal.ex

```diff
@@ -1,13 +1,28 @@
 defmodule AndiWeb.IngestionLiveView.SelectDatasetModal do
   use Phoenix.LiveComponent
 
+  import Ecto.Query, only: [from: 2]
+
+  alias Andi.InputSchemas.Datasets.Dataset
+
+  def mount(socket) do
```

Comment:
Moved all local state that is not needed for API/top level validation into this component. (From the metadata form)

Note: I didn't have enough time to refactor this to it's fullest extent. Technically, this should have it's own UI changeset that plugs into the form, but since the form was already built out with local state (And therefore, no validation), I decided to leave it for time reasons.